metric-schema-java/src/main/java/com/palantir/metric/schema/UtilityGenerator.java

```diff
@@ -82,10 +83,13 @@ static JavaFile generateUtilityClass(
                             Modifier.PRIVATE,
                             Modifier.STATIC,
                             Modifier.FINAL)
-                    .initializer(
-                            "$T.requireNonNullElse($T.class.getPackage().getImplementationVersion(), \"unknown\")",
-                            Objects.class,
-                            className)
+                    .initializer(libraryVersion
+                            .map(version -> CodeBlock.of("$S", version))
```

Comment:
Stared at this for a while before noticing this bit, with the bit below for back compat.

Reply:
I've added some inline comments to make this clearer :-) (I think I've invalidated your approval though)